performance-timeline/idlharness.any.js

```diff
@@ -6,22 +6,44 @@
 
 'use strict';
 
-idl_test(
-  ['performance-timeline'],
-  ['hr-time', 'dom'],
-  idl_array => {
+promise_test(async t => {
+  const observe = new Promise((resolve, reject) => {
     try {
-      const callback = (e, o) => {};
-      self.observerEntry = new ResizeObserverEntry(callback);
+      self.observer = new PerformanceObserver((entries, observer) => {
+        self.entryList = entries;
+        resolve();
+      });
+      observer.observe({ entryTypes: ['mark'] });
+      performance.mark('test');
     } catch (e) {
-      // Will be surfaced when entry is undefined below.
+      reject(e);
     }
+  });
+  const timeout = new Promise((_, reject) => {
+    t.step_timeout(() => reject('Timed out waiting for observation'), 3000);
+  });
+  const execute_test = () => {
+    idl_test(
+      ['performance-timeline'],
+      ['hr-time', 'dom'],
+      idl_array => {
+        idl_array.add_objects({
+          Performance: ['performance'],
+          PerformanceObserver: ['observer'],
+          // NOTE: PerformanceEntry is implicitly tested in user-timing.
+          PerformanceObserverEntryList: ['entryList'],
+          PerformanceObserverEntry: ['entryList[0]'],
```

Comment:
I don't think there's an interface called PerformanceObserverEntry, this is a PerformanceEntry right? Not sure if the name given here matters or not.

Reply:
Yep, the name definitely matters. Corrected the test, by pulling in `user-timing` for a class that extends `PerformanceEntry`. PTAL.